editor/src/components/canvas/dom-walker.ts

```diff
@@ -431,7 +473,7 @@ function getComputedStyle(
   const elementStyle = window.getComputedStyle(element)
   const attributesSetByStylesheet = getCachedAttributesComingFromStyleSheets(
     invalidatedPathsForStylesheetCacheRef,
-    path,
+    paths[0], // TODO is this sufficient to use the first path element for caching?
```

Comment:
If I have
```
const Cake = () => {
  return <div data-uid='cake-root' />
}
const App = () => {
  return <Cake data-uid='app-root' />
}
```
The paths array would contain the path for `'app-root'` and `'cake-root'`, but which would come first?

If the answer is `app-root`, then this should be fine, but if the answer is `cake-root` then I think that would be a problem because it would mean the first path changes as we dig deeper via focusing, so every focus would blow the cache

Reply:
I _think_ we are fine blowing the cache for every refocus. this cache mostly exists to make scrolling speedier, and only re-calculate relevant things while dragging. Actions such as change selection already blow caches in the dom-walker